objects/drawin.c

```diff
@@ -451,6 +452,8 @@ drawin_allocator(lua_State *L)
                           globalconf.default_cmap,
                           xcursor_new(globalconf.cursor_ctx, xcursor_font_fromstr(w->cursor))
                       });
+    if (globalconf.is_compositing)
+        xcb_composite_redirect_subwindows(globalconf.connection, w->window, XCB_COMPOSITE_REDIRECT_MANUAL);
```

Comment:
Good question. I think the composition must happen at the top-level windows (in this case, the drawins). Otherwise the drawin containing the systray widget would still draw the content of the systray window without proper composition.

I tried to use xcb_composite_redirect_window on the systray and that left the black box on the drawin (since I selected the manual update parameter).

Reply:
Okay, thanks for trying.